src/app/services/authentication.service.ts

```diff
@@ -30,14 +31,23 @@ export class AuthenticationService {
 
   login(userData: LoginBody): Observable<any> {
     return this.http.post(this.API.concat('tuiterapi/auth/login'), userData)
-    .map((response: Response) => response.json())
+    .map((response: Response) => response)
     .catch((error: Response) => {
-        return  Observable.throw(error.json());
+        return  Observable.throw(error);
       });
   }
 
   updatePassword(userData: UpdatePassBody) {
-    return this.http.patch(this.API.concat('tuiterapi/users/pass'), userData)
+    const token =  JSON.parse(localStorage.getItem('token'));
```

Comment:
please move this to a method so it can be reused in future

`get getOptions() {
     let token =  JSON.parse(localStorage.getItem('token'));
     let httpOptions = {
           headers: new HttpHeaders({
          'Content-Type':  'application/json',
          'Authorization': token
        })
    };
    return httpOptions;
}`